vcert/connection_tpp.py

```diff
@@ -109,11 +129,220 @@ def auth(self):
+                    key_types = [KeyType(KeyType.RSA, p["KeyPair"]["KeySize"]["Value"])]
+                else:
+                    key_types = [KeyType(KeyType.RSA, x) for x in KeyType.ALLOWED_SIZES]
+            elif p["KeyPair"]["KeyAlgorithm"]["Value"] == "ECC":
+                if p["KeyPair"]["EllipticCurve"]["Locked"]:
+                    key_types = [KeyType(KeyType.ECDSA, p["KeyPair"]["EllipticCurve"]["Value"])]
+                else:
+                    key_types = [KeyType(KeyType.ECDSA, x) for x in KeyType.ALLOWED_CURVES]
+            else:
+                raise ServerUnexptedBehavior
+        else:
+            key_types = []
+            if p["KeyPair"].get("KeySize", {}).get("Locked"):
+                key_types += [KeyType(KeyType.RSA, p["KeyPair"]["KeySize"]["Value"])]
+            else:
+                key_types += [KeyType(KeyType.RSA, x) for x in KeyType.ALLOWED_SIZES]
+            if p["KeyPair"].get("EllipticCurve", {}).get("Locked"):
+                key_types += [KeyType(KeyType.ECDSA, p["KeyPair"]["EllipticCurve"]["Value"])]
+            else:
+                key_types += [KeyType(KeyType.ECDSA, x) for x in KeyType.ALLOWED_CURVES]
+        return Policy(key_types=key_types)
+
+    @staticmethod
+    def _parse_zone_data_to_object(data):
+        s = data["Policy"]["Subject"]
+        ou = s['OrganizationalUnit'].get('Values')
+        policy = TPPConnection._parse_zone_config_to_policy(data)
+        if data["Policy"]["KeyPair"]["KeyAlgorithm"]["Value"] == "RSA":
+            key_type = KeyType(KeyType.RSA, data["Policy"]["KeyPair"]["KeySize"]["Value"])
+        elif data["Policy"]["KeyPair"]["KeyAlgorithm"]["Value"] == "ECC":
+            key_type = KeyType(KeyType.ECDSA, data["Policy"]["KeyPair"]["EllipticCurve"]["Value"])
+        else:
+            key_type = None
+        z = ZoneConfig(
+            organization=CertField(s['Organization']['Value'], locked=s['Organization']['Locked']),
+            organizational_unit=CertField(ou, locked=s['OrganizationalUnit']['Locked']),
+            country=CertField(s['Country']['Value'], locked=s['Country']['Locked']),
+            province=CertField(s['State']['Value'], locked=s['State']['Locked']),
+            locality=CertField(s['City']['Value'], locked=s['City']['Locked']),
+            policy=policy,
+            key_type=key_type,
+        )
+        return z
+
+    def read_zone_conf(self, tag):
+        status, data = self._post(URLS.ZONE_CONFIG, {"PolicyDN":  self._get_policy_dn(tag)})
+        if status != HTTPStatus.OK:
+            raise ServerUnexptedBehavior("Server returns %d status on reading zone configuration." % status)
+        return self._parse_zone_data_to_object(data)
+
     def import_cert(self, request):
         raise NotImplementedError
 
+    @staticmethod
+    def _get_policy_dn(zone):
+        if zone is None:
+            log.error("Bad zone: %s" % zone)
+            raise ClientBadData
+        if re.match(r"^\\\\VED\\\\Policy", zone):
```

Comment:
regex match API would be lot better here and will remove lot of duplicate code below w.r.t regex match